### PR TITLE
fix(ml_vision,ios): add additional null validations

### PR DESCRIPTION
**Pull request by rafaelcmm**
## Description

*This PR is relationed to an error that I checked while using this package on my app. Some scanned barcodes are facing issues with incomplete informations (e.g.: Some driver licenses was returning null on issuedCountry). These null values, when not filled with NSNull, were crashing the app when passing through NSArray addObject. I added a ternary option to driverLicenseDictionary to ensure that null values to be correctly filled*

## Related Issues

*There is no issue related to that. I faced this issue here on my company and fixed it already*

- Fixes https://github.com/FirebaseExtended/flutterfire/issues/4228
- Fixes https://github.com/FirebaseExtended/flutterfire/issues/3506

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] If the pull request affects only one plugin, the PR title starts with the name of the plugin in brackets (e.g. [cloud_firestore])
- [X] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [X] All existing and new tests are passing.
- [X] I updated/added relevant documentation (doc comments with `///`).
- [X] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [X] I read and followed the [Flutter Style Guide].
- [X] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [X] I updated CHANGELOG.md to add a description of the change.
- [X] I signed the [CLA].
- [X] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [X] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/FirebaseExtended/flutterfire/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://www.dartlang.org/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

**Comment by rafaelcmm** [comment]
@googlebot I signed it!

**Comment by rafaelcmm** [comment]
> Would it be possible to add an integration test for this?
> 
> See https://github.com/FirebaseExtended/flutterfire/tree/master/packages/firebase_ml_vision/example/test_driver for examples

@collinjackson Since is a test for Driver Licenses Barcodes, I don't know a good way to test it without explicit add a driver license picture for testing, and I think it is not good to have an image with sensitive information on the project.

Do you know a way so I can test the DL reading feature without have to use a real DL image?

**Comment by rafaelcmm** [comment]
@collinjackson @Ehesp @kroikie @Salakar @bparrishMines Hey fellows, this bugfix is really important for the project my company is working on. 

I would appreciate your help to merge this.

**Review by WilliamCunhaCardoso — APPROVED** [review]

**Review by ypedroo — APPROVED** [review]

**Review by lynrin — APPROVED** [review]
Hey @rafaelcmm. Thank you for watching the PR of #3822.

These changes have worked well for me, and I'm waiting for them to be reviewed by the code owners.

**Comment by rafaelcmm** [comment]
@Ehesp @kroikie @Salakar any progress on this? It seems there is some people needing it.

**Review by WilliamCunhaCardoso — APPROVED** [review]

**Review by uaumateus — APPROVED** [review]

**Comment by jronceroscb** [comment]
Hi @rafaelcmm . We found the same issue with the CALENDAR_EVENT and adding null validation also fixes the problem.

Can you add this null validation to each type that is missing it? So we don't need to push different PR for the same issue.

Thanks!

**Review by joaopereirajp — APPROVED** [review]

**Comment by rafaelcmm** [comment]
> Hi @rafaelcmm . We found the same issue with the CALENDAR_EVENT and adding null validation also fixes the problem.
> 
> Can you add this null validation to each type that is missing it? So we don't need to push different PR for the same issue.
> 
> Thanks!
> 
> EDIT.
> The methods currently missing the null validation are:
> 
> * driverLicenseToDictionary _(in your PR)_
> * calendarEventToDictionary _(tested)_
> * visionBarcodeSMSToDictionary _(not tested)_
> * visionBarcodePhoneToDictionary _(not tested)_
> * visionBarcodeEmailToDictionary _(not tested)_
> * visionBarcodeWiFiToDictionary _(not tested)_
> * visionBarcodeGeoPointToDictionary _(not tested)_

Hey @jronceroscb! I just added the validations you've asked for. Please check it.

**Comment by jronceroscb** [comment]
> > Hi @rafaelcmm . We found the same issue with the CALENDAR_EVENT and adding null validation also fixes the problem.
> > Can you add this null validation to each type that is missing it? So we don't need to push different PR for the same issue.
> > Thanks!
> > EDIT.
> > The methods currently missing the null validation are:
> > 
> > * driverLicenseToDictionary _(in your PR)_
> > * calendarEventToDictionary _(tested)_
> > * visionBarcodeSMSToDictionary _(not tested)_
> > * visionBarcodePhoneToDictionary _(not tested)_
> > * visionBarcodeEmailToDictionary _(not tested)_
> > * visionBarcodeWiFiToDictionary _(not tested)_
> > * visionBarcodeGeoPointToDictionary _(not tested)_
> 
> Hey @jronceroscb! I just added the validations you've asked for. Please check it.

Hello @rafaelcmm ! Thanks for taking the time to add the validations requested. 

By the way, you've also fixed another problem we found in the meantime. There was a typo in the 'organization' field of the contact info that was assigning the field 'jobTitle' instead. 

I hope this PR gets merged soon enough, since we also need this for a production app.

**Comment by rafaelcmm** [comment]
> > > Hi @rafaelcmm . We found the same issue with the CALENDAR_EVENT and adding null validation also fixes the problem.
> > > Can you add this null validation to each type that is missing it? So we don't need to push different PR for the same issue.
> > > Thanks!
> > > EDIT.
> > > The methods currently missing the null validation are:
> > > 
> > > * driverLicenseToDictionary _(in your PR)_
> > > * calendarEventToDictionary _(tested)_
> > > * visionBarcodeSMSToDictionary _(not tested)_
> > > * visionBarcodePhoneToDictionary _(not tested)_
> > > * visionBarcodeEmailToDictionary _(not tested)_
> > > * visionBarcodeWiFiToDictionary _(not tested)_
> > > * visionBarcodeGeoPointToDictionary _(not tested)_
> > 
> > 
> > Hey @jronceroscb! I just added the validations you've asked for. Please check it.
> 
> Hello @rafaelcmm ! Thanks for taking the time to add the validations requested.
> 
> By the way, you've also fixed another problem we found in the meantime. There was a typo in the 'organization' field of the contact info that was assigning the field 'jobTitle' instead.
> 
> I hope this PR gets merged soon enough, since we also need this for a production app.

Yes, I hope so too. I'm maintaining a fork of this project just because of this for a long time, because it breaks our production app too.

Hope any of the code owners check on this. I sent tweets for some of them to please take a look here, but no answer until now.

**Comment by rafaelcmm** [comment]
@collinjackson @Ehesp @kroikie @Salakar @bparrishMines any updates on this?

**Comment by rafaelcmm** [comment]
> @collinjackson @Ehesp @kroikie @Salakar @bparrishMines any updates on this?

@collinjackson @Ehesp @kroikie @Salakar @bparrishMines hey guys, any updates on this PR? I'm keeping  this fork for 2 years now to be able to use the lib on my company project.

There are several people needing it too. Could you guys check it and merge if possible?

File: packages/firebase_ml_vision/ios/firebase_ml_vision.podspec [hunk 1]
```diff
@@ -33,8 +33,8 @@ An SDK that brings Google's machine learning expertise to Android and iOS apps i
   s.source_files = 'Classes/**/*'
   s.public_header_files = 'Classes/**/*.h'
   s.dependency 'Flutter'
-  s.dependency 'firebase_core'
-  s.dependency 'Firebase/MLVision', firebase_sdk_version
+  s.dependency 'Firebase/Core'
+  s.dependency 'Firebase/MLVision'
```
Comment (Salakar) [inline packages/firebase_ml_vision/ios/firebase_ml_vision.podspec:37]: Not sure why these changes were necessary? `firebase_core` is the FlutterFire plugin which transitively should bring in the Firebase/Core dependency? Additionally we use `firebase_sdk_version` to make sure all plugins use the same SDK version

File: packages/firebase_ml_vision/pubspec.yaml [hunk 2]
```diff
@@ -1,7 +1,7 @@
 name: firebase_ml_vision
 description: Flutter plugin for Firebase machine learning vision services.
 homepage: https://github.com/FirebaseExtended/flutterfire/tree/master/packages/firebase_ml_vision
-version: 0.11.0
+version: 0.11.0+1
```
Comment (Salakar) [inline packages/firebase_ml_vision/pubspec.yaml:4]: Versioning is done automatically now, please revert.

File: packages/firebase_ml_vision/CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,11 @@
+## 0.11.0+1
```
Comment (Salakar) [inline packages/firebase_ml_vision/CHANGELOG.md:1]: Changelogs are done automatically now, please revert.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 4]
```diff
@@ -15,7 +15,7 @@ This will ensure a smooth and quick review process. Updating the `pubspec.yaml`
 - [ ] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
 - [ ] All existing and new tests are passing.
 - [ ] I updated/added relevant documentation (doc comments with `///`).
-- [ ] The analyzer (`flutter analyze`) does not report any problems on my PR.
+- [ ] The analyzer (`melos run analyze`) does not report any problems on my PR.
```
Comment (Salakar) [inline .github/PULL_REQUEST_TEMPLATE.md:18]: Please revert this also, we'll sort this separately.

File: packages/firebase_ml_vision/ios/firebase_ml_vision.podspec (hunk 1 above)
Comment (Salakar) [reply packages/firebase_ml_vision/ios/firebase_ml_vision.podspec:37]: Reverting to see what CI does.

**Review by russellwheatley — APPROVED** [review]
Providing the CI passes, this LGTM

**Review by rrousselGit — APPROVED** [review]
